python/ray/experimental/dag/vis_utils.py

```diff
@@ -110,3 +108,42 @@ def plot(dag: DAGNode, to_file=None):
         tmp_file.close()
     except NameError:
         pass
+
+
+class DAGNodeNameGenerator(object):
+    """
+    Generate unique suffix for each given Node in the DAG.
+    Apply monotonic increasing id suffix for duplicated names.
+    """
+
+    def __init__(self):
+        self.name_to_suffix: Dict[str, int] = dict()
+
+    def get_node_name(self, node: DAGNode):
+
+        if isinstance(node, InputNode):
+            node_name = "input_node"
```

Comment:
We actually have an assertion in Ray DAG that it should never have two `InputNode`s, but sadly can only be enforced when we do actual traversal of the DAG via .execute(). It could be slight UX improvement if we ever detect the need to provide a different suffix for `InputNode` with another uuid when user tries to visualize intermediate DAG.

Reply:
Thought about this as I did see the requirement for having only one InputNode.  There are other interesting scenarios there: what if I purposely name my function node as "input_node", lol.  so enforcing it on naming level might not be that useful. But I agree something can be done here to improve UX.